copy/multiple.go

```diff
@@ -301,26 +301,23 @@ func (c *copier) copyMultipleImages(ctx context.Context) (copiedManifest []byte,
 			}
 		}
 
-		// Check if the updates or a type conversion meaningfully changed the list of images
-		// by serializing them both so that we can compare them.
-		attemptedManifestList, err := attemptedList.Serialize()
-		if err != nil {
-			return nil, fmt.Errorf("encoding updated manifest list (%q: %#v): %w", updatedList.MIMEType(), updatedList.Instances(), err)
-		}
-		originalManifestList, err := originalList.Serialize()
-		if err != nil {
-			return nil, fmt.Errorf("encoding original manifest list for comparison (%q: %#v): %w", originalList.MIMEType(), originalList.Instances(), err)
-		}
+		// If possible use the original value instead of the one we just rebuilt,
+		// so that we don't change the digest and keep any additional fields.
+		attemptedManifestList := manifestList
 
+		// Check if the updates or a type conversion meaningfully changed the list of images,
+		// that is if MIME type or the referenced instances changed.
 		// If we can't just use the original value, but we have to change it, flag an error.
-		if !bytes.Equal(attemptedManifestList, originalManifestList) {
+		if attemptedList.MIMEType() != originalList.MIMEType() || !reflect.DeepEqual(attemptedList.Instances(), originalList.Instances()) {
```

Comment:
This condition is incorrect in general; e.g. layer annotation changes on the `attemptedList` would not change `.Instances()`. We do need to compare _everything_.

Reply:
I might be missing something but I don't see anything in `copyMultipleImages` that would edit the annotations.

Reply:
That might easily change in the future, this high-level code has no reason to assume it understands everything about the format..